src/test/java/io/github/jhipster/online/service/JdlServiceTest.java

```diff
@@ -0,0 +1,196 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import io.github.jhipster.online.config.ApplicationProperties;
+import io.github.jhipster.online.domain.Jdl;
+import io.github.jhipster.online.domain.JdlMetadata;
+import io.github.jhipster.online.domain.User;
+import io.github.jhipster.online.domain.enums.GitProvider;
+import io.github.jhipster.online.repository.JdlRepository;
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@ExtendWith(SpringExtension.class)
+@EnableConfigurationProperties(value = ApplicationProperties.class)
+@TestPropertySource("classpath:config/application.yml")
+class JdlServiceTest {
+
+    @Mock
+    private LogsService logsService;
+
+    @Mock
+    private GitService gitService;
+
+    @Mock
+    private JHipsterService jHipsterService;
+
+    @Mock
+    private GithubService githubService;
+
+    @Mock
+    private GitlabService gitlabService;
+
+    @Mock
+    private JdlRepository jdlRepository;
+
+    @Autowired
+    private ApplicationProperties applicationProperties;
+
+    private JdlService jdlService;
+
+    @BeforeEach
+    void shouldConstructJHipsterService() {
+        jdlService =
+            new JdlService(logsService, gitService, jHipsterService, githubService, gitlabService, jdlRepository, applicationProperties);
+    }
+
+    @Test
+    void applyJdlWithException() {
+        final User user = new User();
+        final String applyJdlId = "apply-id";
+        final JdlMetadata jdlMetadata = new JdlMetadata();
+
+        jdlService.applyJdl(user, "jhipster", "online", jdlMetadata, applyJdlId, GitProvider.GITHUB);
+
+        verify(logsService).addLog(applyJdlId, "Error during generation: Error creating file jhipster-jdl.jh, could not write the file");
+        verify(logsService).addLog(applyJdlId, "Generation failed");
+    }
+
+    @Test
+    void applyGithubJdl() throws GitAPIException, IOException {
+        final User user = new User();
+        final String applyJdlId = "apply-id";
+        final JdlMetadata jdlMetadata = new JdlMetadata();
+        jdlMetadata.setId("1");
+        Jdl jdl = mock(Jdl.class);
+        jdlMetadata.setName("jdl-metadata-name");
+        given(jdlRepository.findOneByJdlMetadataId(jdlMetadata.getId())).willReturn(Optional.of(jdl));
+        given(githubService.getHost()).willReturn("https://www.github.com");
```

Comment:
I would use the property `application.github.host` from `application.yml` instead of hardcoding `https://www.github.com`

Reply:
Done

Reply:
@roexber : Not sure this is done, as I don't see the property `application.github.host` being used. Also we need to add that property to the `application.yml` file as it's not there; https://github.com/jhipster/jhipster-online/blob/8fe29ce26e0eaaa6d4574cac57144ece9d995d2d/src/test/resources/config/application.yml#L117. So ultimately we would have, 

```
application:
    github:
        host: https://github.com
    gitlab:
        host: https://gitlab.com
```

in the `application.yml` file and then you could refer to that using `.willReturn(applicationProperties.getGithub().getHost())`.

Reply:
@SudharakaP due to

@EnableConfigurationProperties(value = ApplicationProperties.class)
@TestPropertySource("classpath:config/application.yml")

on the test class and 
    @Autowired
    private ApplicationProperties applicationProperties;

the properties are loaded from
src/test/resources/config/application.yml

I'm not explicitly overriding since there are sensible defaults set on io.github.jhipster.online.config.ApplicationProperties Github and Gitlab child classes


    public static class Github {

        private String host = "https://github.com";

Testing if spring property loading works seemed out of scope, I'll override them anyway to make sure the default is substituted.

Reply:
Tried adding 

application:
  github:
    host: github.test
  gitlab:
    host: gitlab.test

to the src/test/resources/config/application.yml without much luck... not sure how to go about it next

Reply:
Managed to figure this one out using the help of @patrickhancke